src/server/mod.rs

```diff
@@ -18,12 +21,42 @@ pub use encode::Encoder;
 pub struct ServerOptions {
     /// Timeout to handle headers. Defaults to 60s.
     headers_timeout: Option<Duration>,
+    default_host: Option<String>,
+}
+
+impl ServerOptions {
+    /// constructs a new ServerOptions with default settings
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// sets the timeout by which the headers must have been received
+    pub fn with_headers_timeout(mut self, headers_timeout: Duration) -> Self {
+        self.headers_timeout = Some(headers_timeout);
+        self
+    }
+
+    /// Sets the default http 1.0 host for this server. If no host
+    /// header is provided on an http/1.0 request, this host will be
+    /// used to construct the Request Url.
+    ///
+    /// If this is not provided, the server will respond to all
+    /// http/1.0 requests with status `505 http version not
+    /// supported`, whether or not a host header is provided.
```

Comment:
:100: -- I really like how this works! This can very naturally translate to a CLI flag which sets the default URL, or a config which mandates the hostname is set. This feels exactly right.